src/tree/cache.rs

```diff
@@ -26,6 +26,8 @@ pub struct Cache {
     final_layout_entry: Option<CacheEntry<LayoutOutput>>,
     /// The cache entries for the node's preliminary size measurements
     measure_entries: [Option<CacheEntry<Size<f32>>>; CACHE_SIZE],
+    /// Marks if all cache entries have been cleared
+    cleared: bool,
```

Comment:
Nit: could we call this `is_empty`. As it will also be true when a node is first initialised, at which point it has not technically been cleared.